### PR TITLE
feat(store): implement updateItem operator

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #545 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by arturovt** [comment]
@markwhitfeld i will rebase and remove duplicate code after #780 is merged :rocket:

**Comment by markwhitfeld** [comment]
#780 is merged

File: packages/store/operators/tests/update-item.spec.ts [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+        a: updateItem(undefined!, {})
+      })(original);
+
+      // Assert
+      expect(newValue).toBe(original);
+    });
+  });
+
+  describe('when non-existing index or predicate provided', () => {
+    it('returns the same root if non-existing index provided', () => {
```
Comment (markwhitfeld) [inline packages/store/operators/tests/update-item.spec.ts]: This should rather just set the item IMO.

**Comment by arturovt** [comment]
@markwhitfeld ping :rocket:

**Pull request merged**